src/drivers/lights/rgbled_lp5562/rgbled_lp5562.cpp

```diff
@@ -0,0 +1,320 @@
+
+	LedController _led_controller;
+	uint8_t _current = 175; // matching default current of 17.5mA
+};
+
+RGBLED_LP5562::RGBLED_LP5562(const I2CSPIDriverConfig &config) :
+	I2C(config),
+	I2CSPIDriver(config)
+{
+	_current = config.custom1;
+}
+
+int
+RGBLED_LP5562::init()
+{
+	int ret = I2C::init();
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	uint8_t command[1] = {ENABLE_CHIP_EN};
+	ret = write(ENABLE_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	// We have to wait 500us after enable.
+	px4_usleep(500);
+
+	command[0] = CONFIG_ENABLE_INTERNAL_CLOCK;
+	ret = write(CONFIG_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	command[0] = LED_MAP_ALL_PWM;
+	ret = write(LED_MAP_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	// Write all 3 colors at once.
+	uint8_t currents[3] = {_current, _current, _current};
+	ret = write(B_CURRENT_ADDR, currents, sizeof(currents));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	ScheduleNow();
+
+	return OK;
+}
+
+int
+RGBLED_LP5562::probe()
+{
+	uint8_t result[1] = {0};
+	int ret = read(W_CURRENT_ADDR, result, sizeof(result));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	_retries = 1;
+
+	return (result[0] == W_CURRENT_DEFAULT) ? OK : ERROR;
+}
+
+int
+RGBLED_LP5562::read(uint8_t address, uint8_t *data, unsigned count)
+{
+	uint8_t cmd = address;
+	return transfer(&cmd, 1, (uint8_t *)data, count);
+}
+
+int
+RGBLED_LP5562::write(uint8_t address, uint8_t *data, unsigned count)
+{
+	uint8_t buf[4];
+
+	if (sizeof(buf) < (count + 1)) {
+		return -EIO;
+	}
+
+	buf[0] = address;
+	memcpy(&buf[1], data, count);
+
+	return transfer(&buf[0], count + 1, nullptr, 0);
+}
+
+void
+RGBLED_LP5562::RunImpl()
+{
+	if (should_exit()) {
+		send_led_rgb(0, 0, 0);
+		return;
+	}
+
+	LedControlData led_control_data;
+
+	if (_led_controller.update(led_control_data) == 1) {
+
+		const uint8_t on = led_control_data.leds[0].brightness;
+
+		switch (led_control_data.leds[0].color) {
+		case led_control_s::COLOR_RED:
+			send_led_rgb(on, 0, 0);
+			break;
+
+		case led_control_s::COLOR_GREEN:
+			send_led_rgb(0, on, 0);
+			break;
+
+		case led_control_s::COLOR_BLUE:
+			send_led_rgb(0, 0, on);
+			break;
+
+		case led_control_s::COLOR_AMBER: // same as yellow
+		case led_control_s::COLOR_YELLOW:
+			send_led_rgb(on, on, 0);
+			break;
+
+		case led_control_s::COLOR_PURPLE:
+			send_led_rgb(on, 0, on);
+			break;
+
+		case led_control_s::COLOR_CYAN:
+			send_led_rgb(0, on, on);
+			break;
+
+		case led_control_s::COLOR_WHITE:
+			send_led_rgb(on, on, on);
+			break;
+
+		case led_control_s::COLOR_OFF:
+		default:
+			send_led_rgb(0, 0, 0);
+			break;
+		}
+
+	}
+
+	ScheduleDelayed(_led_controller.maximum_update_interval());
+}
+
+int
+RGBLED_LP5562::send_led_rgb(uint8_t r, uint8_t g, uint8_t b)
+{
+	uint8_t leds[3] = {b, g, r};
+	return write(B_PWM_ADDR, leds, sizeof(leds));
+}
+
+void
+RGBLED_LP5562::print_usage()
```

Comment:
@julianoes What's the build error?

A bigger problem than this PR, but the docs in these kinds of driver docs are pretty useless unless you already know what the driver is. For example, see how something like this renders: https://docs.px4.io/main/en/modules/modules_driver.html#rgbled

I'm trying to prototype at least some useful documentation in https://github.com/PX4/PX4-Autopilot/pull/21638. Essentially this means addition of a description that explains what the driver is, when you need to install it in firmware and how, how you enable it, any params you need, maybe a manufacturer URL to make it easy to find more information.

Thoughts?

Reply:
Thanks for the hint. It turns out I can't have nice things. `I2C_ADDR` needs to be a `define` and not a `static constexpr` variable. Fine :disappointed:.

Yes, actual instructions how to use it would be more useful, I don't disagree. Not something I can address right now though.

Reply:
If not now, when? https://github.com/PX4/PX4-Autopilot/pull/21649/files#r1212458987

Reply:
This will automatically work if you connect a Holybro GPS that includes this RGB driver.

Reply:
@hamishwillee does that answer it, or what else would you note in that case?